Rules exceptions subfeatures

dhurley14 opened the pull request:
## Summary

Summarize your PR. If it involves visual changes include a screenshot or gif.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

rylnd committed:
Introduces Rules feature and subsequent app changes

This does not include changes to existing roles, nor the role migration
machinery.

rylnd committed:
style: revert linting changes to large yaml/JSON files

These changes were made automatically in an initial commit that added
our new features to roles; those changes have since been reverted
(320c34f485), and thus there should not currently be any behavioral
changes in these files, which makes these stylistic changes even more
unnecessary.

Note: I also noticed that a few old references had (accidentally?)
remained in `security_roles.json` after `320c34f485`; this cleans those
up as well.

denar50 committed:
update the siem migrations required permissions

Instead of requiring siemVX read/all, it now requires securitySolutionRulesV1 read/all

denar50 committed:
update the siem migrations required permissions

Instead of requiring siemVX read/all, it now requires securitySolutionRulesV1 read/all

denar50 committed:
update attack discovery required permissions

denar50 committed:
update timeline required permissions

logeekal committed:
fix: siem migration privs

logeekal committed:
fix: tests

denar50 committed:
fix onboarding sections

It is unclear on wether "dashboards" and "integrations" should be exclusive to `siemV5` or `securitySolutionRulesV1`. So for now we are showing it when the user has either of those.

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

denar50 committed:
allow alert management operations for roles with read only access

This is the current behaviour accoring do the documentation https://www.elastic.co/docs/solutions/security/detect-and-alert/detections-requirements.

denar50 committed:
fix create shared exception list endpoint

Now it requires the `securitySolutionRulesV1.all` privilege

denar50 committed:
fix unit test

dplumlee committed:
fixes bulk_action route to allow export on rules-read

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

denar50 committed:
fix cypress test: endpoint_role_rbac_with_space_awareness.cy.ts

No security subfeature is required in all spaces anymore. The test was failing because the `siemV5` feature file never got updated and it was still referencing a feature flag that has been enabled and removed in `main`.
The feature flag in question is `endpointManagementSpaceAwarenessEnabled` which was being used to override the subfeature configuration by setting `requireAllSpaces=false` and `privilegesTooltip=undefined`. Now that the feature flag doesn't exist, it makes sense to remove these properties directly in the subfeature configuration instead of overriding them outside of it.

denar50 committed:
Merge branch 'main' into rules-rbac-new

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

denar50 committed:
fix manage value lists button disabled for rules-all

The logic to show it was relying on the old siemPrivileges, however value lists is now under rules.

dplumlee committed:
fix showing rule update callouts when users don't have rules-all

denar50 committed:
fix authorization tests

Reshuffling privileges and removal of alerting privileges from siemV5. These alerting privileges exist exclusively in securitySolutionRulesV1

denar50 committed:
Merge branch 'main' into rules-rbac-new

rylnd committed:
Merge branch 'main' into rules-rbac-new

This notably includes the fix to the infinite loop on the alerts page
when a role lacks sufficient lists privileges.

denar50 committed:
Merge remote-tracking branch 'upstream/main' into rules-rbac-1

denar50 committed:
add missing socManagement sub feature config to siemV4 and siemV5

denar50 committed:
fix ai4soc cypress test
The test broke after merging main into the branch

denar50 committed:
add missing siemV5 case to trusted devices rbac cypress test

denar50 committed:
fix authorization tests after new ai4soc changes

denar50 committed:
fix api privileges tests after ai4soc changes

dplumlee committed:
fixes respectLicenseLevel test that broke merging upstream

denar50 committed:
Merge branch 'main' into rules-rbac-new

denar50 committed:
add cypress tests for the rules management page

dplumlee committed:
removes unused code and resolves PR TODOs

denar50 committed:
update SECURITY_FEATURE_ID constant to siemV5

denar50 committed:
Merge branch 'main' into rules-rbac-new

denar50 committed:
fix rules management page privileges cypress test failure

dplumlee committed:
removes remaining TODOs and unskips tests

denar50 committed:
add cypress tests for the alerts page

denar50 committed:
Merge branch 'main' into rules-rbac-new

denar50 committed:
add cypress tests for the rule details page

dplumlee committed:
Merge branch 'main' into rules-rbac-new

dplumlee committed:
switches detection engine health setup endpiont to have initalize-sec-sol permission

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

dplumlee committed:
Merge remote-tracking branch 'upstream/main' into rules-rbac-new

dplumlee committed:
removes commented out code

dhurley14 committed:
adds deprecation / migration from rulev1 to rulev2 and init exceptions subfeature

dplumlee committed:
fixes attack page merge conflicts

dhurley14 committed:
Merge remote-tracking branch 'rylnd/rules-rbac-new' into rules-exceptions-subfeatures

dhurley14 committed:
typo lead to bug in UI where buttons were disabled

rylnd committed:
Merge branch 'main' into rules-rbac-new

dhurley14 committed:
continue registering rules v1 alongside rules v2, use alerting rules client function for create rule-exception route

rylnd committed:
Allow users with Dashboard privileges to view the Automatic Migrations section

Security:read users should be able to view this section.

Co-authored-by: Jatin Kathuria <jtn.kathuria@gmail.com>

rylnd committed:
Update new Attacks links from upstream to use new Rules privileges

This is new code since the RBAC branch was opened, so we need to make
the appropriate updates here.

NB that we are currently assuming that Attack Alerts will use Alerts
privileges, and thus are making these links require the Rules-based UI
privileges, which will later be refined to alerts-specific privileges.

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

rylnd committed:
Merge branch 'main' into rules-rbac-new

 Conflicts:
	x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/index.test.tsx

rylnd committed:
Merge branch 'main' into rules-rbac-new

 Conflicts:
	x-pack/solutions/security/packages/features/src/security/v1_features/kibana_features.ts
	x-pack/solutions/security/packages/features/src/security/v4_features/kibana_features.ts

dhurley14 committed:
Merge remote-tracking branch 'rylnd/rules-rbac-new' into rules-exceptions-subfeatures

dplumlee committed:
adds exceptions permissions to user privileges hook

dhurley14 committed:
assign exception so management to subfeature

dhurley14 committed:
Merge remote-tracking branch 'origin/rules-exceptions-subfeatures' into rules-exceptions-subfeatures

rylnd committed:
Merge branch 'main' into rules-rbac-new

rylnd committed:
Merge branch 'main' into rules-rbac-new

 Conflicts:
	x-pack/solutions/security/plugins/security_solution/public/common/components/header_actions/actions.test.tsx

rylnd committed:
fix: privilege requirements for Migrations components

* Refactors MigrationServiceBase to defer certain method implementations
  to child classes
* Migration Service does not _require_ `rules:read`, but it is an
  optional capability
* Dashboard Migrations depend on `dashboard:show` AND `siemv5:show`
* Rule Migrations depend on `rules:read`
* Migrations Landing page depends on `rules:read` OR `siemv5:show`

rylnd committed:
Fix types in test class implementation

The previous commit made these part of the abstract class, so we need to
define them here.

logeekal committed:
Merge branch 'rules-rbac-new' into fix/automatic_migrations_rules_rbac

logeekal committed:
fix: dashboard and rules privs

logeekal committed:
revert unncessary change

dplumlee committed:
lets shared exception list page be viewed with no rules/endpoint exceptions

dplumlee committed:
Merge remote-tracking branch 'rylnd/rules-rbac-new' into rules-exceptions-subfeatures

dhurley14 committed:
updates routes with exception subfeature authz strings

dhurley14 committed:
Merge remote-tracking branch 'origin/rules-exceptions-subfeatures' into rules-exceptions-subfeatures

dhurley14 committed:
remove exceptions api authz strings from rules v2 feature, should only be granted via exceptions subfeature

rylnd committed:
Merge branch 'main' into rules-rbac-new

 Conflicts:
	x-pack/solutions/security/plugins/lists/server/routes/list_index/create_list_index_route.ts

dhurley14 committed:
linking rules should use the edit + read rules authz function

dplumlee committed:
updates shared list UI page to handle RBAC edge cases

rylnd committed:
Merge branch 'main' into rules-rbac-new

kibanamachine committed:
Changes from node scripts/regenerate_moon_projects.js --update

logeekal committed:
fix: tests + types

dplumlee committed:
refactors feature config code to match other features

dplumlee committed:
Merge remote-tracking branch 'rylnd/rules-rbac-new' into rules-exceptions-subfeatures

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/365340)
* Commit: 513e9743b303e994105379b55764656d0b107bd9

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/365340#019abc6a-1563-44f7-858a-d161d25fad19)

cc @dhurley14 @dplumlee

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/365340","number":365340,"commit":"513e9743b303e994105379b55764656d0b107bd9"}],"number":365340}
buildkite-pr-comment-->

dplumlee committed:
fix types

dplumlee committed:
updates server features mock

rylnd committed:
Merge pull request #13 from logeekal/fix/automatic_migrations_rules_rbac

[Automatic Migrations] Fix Automatic Migrations permissions for Rules RBAC

rylnd committed:
Simplify permissions for migration dashboard link

This feature only cares about the dashboard and Migrations feature
privileges; while intermediate privileges may not have any effect on
this behavior, they are an unnecessary complication and make this
feature more inflexible in the future.

dplumlee committed:
update product features service mock

dplumlee committed:
Merge remote-tracking branch 'rylnd/rules-rbac-new' into rules-exceptions-subfeatures

dplumlee committed:
fix exception list jest tests

dplumlee committed:
fixes take action dropdown jest tests

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/366158)
* Commit: e77cd1c754d012f4ab8424ec46dbd58b7939ebec
* [Cloud Deployment](https://kibana-pr-243095-3e3cbd.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #366058](https://buildkite.com/elastic/kibana-pull-request/builds/366058) failed 12c778144aadaf0b2f3e537855c54df7c28909f1
* :broken_heart: [Build #365464](https://buildkite.com/elastic/kibana-pull-request/builds/365464) failed 635e19501f4c069801ad96c8025df7289975543d
* :broken_heart: [Build #365409](https://buildkite.com/elastic/kibana-pull-request/builds/365409) failed 65795596518c9a009bcae921f792b55758a2cdee
* :broken_heart: [Build #365380](https://buildkite.com/elastic/kibana-pull-request/builds/365380) failed ee43150601e2fc4cfaa2895fb16ca146dda6625b
* :broken_heart: [Build #365340](https://buildkite.com/elastic/kibana-pull-request/builds/365340) failed 513e9743b303e994105379b55764656d0b107bd9

cc @dhurley14 @dplumlee

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366158","number":366158,"commit":"e77cd1c754d012f4ab8424ec46dbd58b7939ebec"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366058","number":366058,"commit":"12c778144aadaf0b2f3e537855c54df7c28909f1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/365464","number":365464,"commit":"635e19501f4c069801ad96c8025df7289975543d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/365409","number":365409,"commit":"65795596518c9a009bcae921f792b55758a2cdee"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/365380","number":365380,"commit":"ee43150601e2fc4cfaa2895fb16ca146dda6625b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/365340","number":365340,"commit":"513e9743b303e994105379b55764656d0b107bd9"}],"number":366158}
buildkite-pr-comment-->

denar50 committed:
Merge branch 'main' into rules-rbac-new

dhurley14 committed:
applies patch rules using new alerting bulk edit rules function if all fields are part of the read authz editable fields

dhurley14 committed:
Merge remote-tracking branch 'origin/rules-exceptions-subfeatures' into rules-exceptions-subfeatures

logeekal committed:
fix: privs

logeekal committed:
remove unnecessary file

rylnd committed:
Merge branch 'main' into rules-rbac-new

 Conflicts:
	x-pack/solutions/security/test/serverless/api_integration/test_suites/platform_security/authorization.ts

rylnd committed:
Merge pull request #15 from logeekal/fix/automatic_migration_rbac_fix_server

[Automatic Migrations] Fixed Privileges for Dashboard endpoints.

rylnd committed:
Grant 'alerting.alert.all' privileges to rules:read users

In order to maintain backwards compatibility with alert privileges for
existing SIEM users, we need more than `alert.read` for `siem:read`
users`. Since our only other option is `alert.all`, we instead grant
them this privilege (although arguably this is more access than they
previously had). However, once the `securityAlerts` privilege work is
complete, `securityAlerts.all` will be pared down to precisely what is
needed for security users.

Discussion around this change can be found
[here](https://github.com/elastic/kibana/pull/239634/files/632bbc0e8c2597ae9f676123d6ecd09b644bbf73#r2510764307).

rylnd committed:
Add missing Defend privileges to siemV5

These were deleted before siemv5 was defined, then added back since.
Luckily we have tests verifying the fact that these privileges are
"replaced."

dplumlee committed:
hides exceptions tab in rule details based on permissions

gergoabraham committed:
handle siem version dynamically in  artifact details tab cypress test

dplumlee committed:
fixes search bar jest tests

rylnd committed:
Merge pull request #16 from gergoabraham/fix-artifact-details-cypress-for-siem-upgrade

[Defend Workflows cypress fix] Handle siem version dynamically in  artifact details tab cypress test

dplumlee committed:
updates api integration configs

dplumlee committed:
updates serverless test authorization snapshot

dplumlee committed:
Merge remote-tracking branch 'rylnd/rules-rbac-new' into rules-exceptions-subfeatures

dplumlee committed:
Merge branch 'main' into rules-rbac-new

dplumlee committed:
Merge remote-tracking branch 'rylnd/rules-rbac-new' into rules-exceptions-subfeatures

dplumlee committed:
fix types

dplumlee committed:
Merge remote-tracking branch 'rylnd/rules-rbac-new' into rules-exceptions-subfeatures

dplumlee committed:
fix security role object types

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/368013)
* Commit: bf24ee0ac3f1153e231cd37d9a481bafd4102db2

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/368013#019ae10f-5895-4f95-bcd9-f9bfa65b832a)

### History
* :broken_heart: [Build #368005](https://buildkite.com/elastic/kibana-pull-request/builds/368005) failed dcc9d74eff2e907cb4f8530c21005a4c0f7695bc
* :broken_heart: [Build #367972](https://buildkite.com/elastic/kibana-pull-request/builds/367972) failed d5bcf47b0f22280c9085aa4f06a1092ad46f7900
* :broken_heart: [Build #366158](https://buildkite.com/elastic/kibana-pull-request/builds/366158) failed e77cd1c754d012f4ab8424ec46dbd58b7939ebec
* :broken_heart: [Build #366058](https://buildkite.com/elastic/kibana-pull-request/builds/366058) failed 12c778144aadaf0b2f3e537855c54df7c28909f1
* :broken_heart: [Build #365464](https://buildkite.com/elastic/kibana-pull-request/builds/365464) failed 635e19501f4c069801ad96c8025df7289975543d
* :broken_heart: [Build #365409](https://buildkite.com/elastic/kibana-pull-request/builds/365409) failed 65795596518c9a009bcae921f792b55758a2cdee

cc @dhurley14 @dplumlee

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368013","number":368013,"commit":"bf24ee0ac3f1153e231cd37d9a481bafd4102db2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368005","number":368005,"commit":"dcc9d74eff2e907cb4f8530c21005a4c0f7695bc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367972","number":367972,"commit":"d5bcf47b0f22280c9085aa4f06a1092ad46f7900"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366158","number":366158,"commit":"e77cd1c754d012f4ab8424ec46dbd58b7939ebec"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366058","number":366058,"commit":"12c778144aadaf0b2f3e537855c54df7c28909f1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/365464","number":365464,"commit":"635e19501f4c069801ad96c8025df7289975543d"}],"number":368013}
buildkite-pr-comment-->

dplumlee committed:
REMOVES exceptions serverless security_roles for now

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/368024)
* Commit: 7a74953bf1cf2f9ad34b41dcfbb0cc4cf3318544

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/368024#019ae13c-235e-4c78-8ca4-c76f03bee69d)

### History
* :broken_heart: [Build #368013](https://buildkite.com/elastic/kibana-pull-request/builds/368013) failed bf24ee0ac3f1153e231cd37d9a481bafd4102db2
* :broken_heart: [Build #368005](https://buildkite.com/elastic/kibana-pull-request/builds/368005) failed dcc9d74eff2e907cb4f8530c21005a4c0f7695bc
* :broken_heart: [Build #367972](https://buildkite.com/elastic/kibana-pull-request/builds/367972) failed d5bcf47b0f22280c9085aa4f06a1092ad46f7900
* :broken_heart: [Build #366158](https://buildkite.com/elastic/kibana-pull-request/builds/366158) failed e77cd1c754d012f4ab8424ec46dbd58b7939ebec
* :broken_heart: [Build #366058](https://buildkite.com/elastic/kibana-pull-request/builds/366058) failed 12c778144aadaf0b2f3e537855c54df7c28909f1

cc @dhurley14 @dplumlee

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368024","number":368024,"commit":"7a74953bf1cf2f9ad34b41dcfbb0cc4cf3318544"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368013","number":368013,"commit":"bf24ee0ac3f1153e231cd37d9a481bafd4102db2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368005","number":368005,"commit":"dcc9d74eff2e907cb4f8530c21005a4c0f7695bc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367972","number":367972,"commit":"d5bcf47b0f22280c9085aa4f06a1092ad46f7900"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366158","number":366158,"commit":"e77cd1c754d012f4ab8424ec46dbd58b7939ebec"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366058","number":366058,"commit":"12c778144aadaf0b2f3e537855c54df7c28909f1"}],"number":368024}
buildkite-pr-comment-->

dplumlee committed:
reverts security roles changes

dplumlee committed:
fix types

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/368061)
* Commit: a304a999a32d05c57ca43d87fce4fb0d3dcfce7a
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-243095)
* [Cloud Deployment](https://kibana-pr-243095-27972e.kb.us-west2.gcp.elastic-cloud.com)

### Failed CI Steps
* [Detection Engine - Exceptions - Security Solution Cypress Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/368061#019ae292-ef85-4394-a06c-4cf17839dd30)
* [Rule Management - Security Solution Cypress Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/368061#019ae292-ef92-4eb5-acee-8cc0adbe6a55)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/368061#019ae292-ef92-4eb5-acee-8cc0adbe6a55) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/368061/jobs/019ae292-ef92-4eb5-acee-8cc0adbe6a55/artifacts/019ae2c4-a8d0-4208-80d3-b91b57836a8f) Rule Management - Security Solution Cypress Tests #<span></span>3 / Rules table - privileges securitySolutionRulesV1.all "before all" hook for "should be able to edit rules" "before all" hook for "should be able to edit rules"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/368061#019ae292-ef85-4394-a06c-4cf17839dd30) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/368061/jobs/019ae292-ef85-4394-a06c-4cf17839dd30/artifacts/019ae2c0-f56b-48fd-b699-06708f85c749) Detection Engine - Exceptions - Security Solution Cypress Tests #<span></span>2 / Shared exception lists - read only Exception list actions should be disabled Exception list actions should be disabled

### History
* :broken_heart: [Build #368024](https://buildkite.com/elastic/kibana-pull-request/builds/368024) failed 7a74953bf1cf2f9ad34b41dcfbb0cc4cf3318544
* :broken_heart: [Build #368013](https://buildkite.com/elastic/kibana-pull-request/builds/368013) failed bf24ee0ac3f1153e231cd37d9a481bafd4102db2
* :broken_heart: [Build #368005](https://buildkite.com/elastic/kibana-pull-request/builds/368005) failed dcc9d74eff2e907cb4f8530c21005a4c0f7695bc
* :broken_heart: [Build #367972](https://buildkite.com/elastic/kibana-pull-request/builds/367972) failed d5bcf47b0f22280c9085aa4f06a1092ad46f7900

cc @dhurley14 @dplumlee

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368061","number":368061,"commit":"a304a999a32d05c57ca43d87fce4fb0d3dcfce7a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368024","number":368024,"commit":"7a74953bf1cf2f9ad34b41dcfbb0cc4cf3318544"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368013","number":368013,"commit":"bf24ee0ac3f1153e231cd37d9a481bafd4102db2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368005","number":368005,"commit":"dcc9d74eff2e907cb4f8530c21005a4c0f7695bc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367972","number":367972,"commit":"d5bcf47b0f22280c9085aa4f06a1092ad46f7900"}],"number":368061}
buildkite-pr-comment-->

dplumlee committed:
update deprecated feature ID FTR tests

dplumlee committed:
updates roles for rule cypress tests

rylnd committed:
Merge branch 'main' into rules-rbac-new

dplumlee committed:
allows users with read privileges to view uninstalled prebuilt rules and updateable rules

dplumlee committed:
adds mocks to rule upgrade tests

dplumlee committed:
updates rule upgrade cypress tests

dplumlee committed:
Merge remote-tracking branch 'rylnd/rules-rbac-new' into rules-exceptions-subfeatures

rylnd committed:
Add SIEM V4 user to AI4DSOC capabilities tests

This will ensure behaviors are correct for all intermediate SIEM
features.

dplumlee committed:
fix rules management cypress tests

dplumlee committed:
Merge remote-tracking branch 'rylnd/rules-rbac-new' into rules-exceptions-subfeatures

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/368785)
* Commit: 9c2a81357a42ca39e92e65a91d7696415b6f8a33

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/368785#019ae70b-29bc-40e2-81b0-47e47b223fa1)

### History
* :broken_heart: [Build #368505](https://buildkite.com/elastic/kibana-pull-request/builds/368505) failed 33f8eae674578a3ab6a86d580e33a8ee4a9b281e
* :broken_heart: [Build #368061](https://buildkite.com/elastic/kibana-pull-request/builds/368061) failed a304a999a32d05c57ca43d87fce4fb0d3dcfce7a
* :broken_heart: [Build #368024](https://buildkite.com/elastic/kibana-pull-request/builds/368024) failed 7a74953bf1cf2f9ad34b41dcfbb0cc4cf3318544
* :broken_heart: [Build #368013](https://buildkite.com/elastic/kibana-pull-request/builds/368013) failed bf24ee0ac3f1153e231cd37d9a481bafd4102db2
* :broken_heart: [Build #368005](https://buildkite.com/elastic/kibana-pull-request/builds/368005) failed dcc9d74eff2e907cb4f8530c21005a4c0f7695bc

cc @dhurley14 @dplumlee

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368785","number":368785,"commit":"9c2a81357a42ca39e92e65a91d7696415b6f8a33"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368505","number":368505,"commit":"33f8eae674578a3ab6a86d580e33a8ee4a9b281e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368061","number":368061,"commit":"a304a999a32d05c57ca43d87fce4fb0d3dcfce7a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368024","number":368024,"commit":"7a74953bf1cf2f9ad34b41dcfbb0cc4cf3318544"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368013","number":368013,"commit":"bf24ee0ac3f1153e231cd37d9a481bafd4102db2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368005","number":368005,"commit":"dcc9d74eff2e907cb4f8530c21005a4c0f7695bc"}],"number":368785}
buildkite-pr-comment-->

dplumlee committed:
fix types

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/368788)
* Commit: 2f1bc7703937686be8b59806420eadd9762e41ac
* [Cloud Deployment](https://kibana-pr-243095-27972e.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #368785](https://buildkite.com/elastic/kibana-pull-request/builds/368785) failed 9c2a81357a42ca39e92e65a91d7696415b6f8a33
* :broken_heart: [Build #368505](https://buildkite.com/elastic/kibana-pull-request/builds/368505) failed 33f8eae674578a3ab6a86d580e33a8ee4a9b281e
* :broken_heart: [Build #368061](https://buildkite.com/elastic/kibana-pull-request/builds/368061) failed a304a999a32d05c57ca43d87fce4fb0d3dcfce7a
* :broken_heart: [Build #368024](https://buildkite.com/elastic/kibana-pull-request/builds/368024) failed 7a74953bf1cf2f9ad34b41dcfbb0cc4cf3318544
* :broken_heart: [Build #368013](https://buildkite.com/elastic/kibana-pull-request/builds/368013) failed bf24ee0ac3f1153e231cd37d9a481bafd4102db2
* :broken_heart: [Build #368005](https://buildkite.com/elastic/kibana-pull-request/builds/368005) failed dcc9d74eff2e907cb4f8530c21005a4c0f7695bc

cc @dhurley14 @dplumlee

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368788","number":368788,"commit":"2f1bc7703937686be8b59806420eadd9762e41ac"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368785","number":368785,"commit":"9c2a81357a42ca39e92e65a91d7696415b6f8a33"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368505","number":368505,"commit":"33f8eae674578a3ab6a86d580e33a8ee4a9b281e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368061","number":368061,"commit":"a304a999a32d05c57ca43d87fce4fb0d3dcfce7a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368024","number":368024,"commit":"7a74953bf1cf2f9ad34b41dcfbb0cc4cf3318544"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368013","number":368013,"commit":"bf24ee0ac3f1153e231cd37d9a481bafd4102db2"}],"number":368788}
buildkite-pr-comment-->

dplumlee committed:
Merge remote-tracking branch 'upstream/main' into rules-rbac-new

dplumlee committed:
Merge remote-tracking branch 'rylnd/rules-rbac-new' into rules-exceptions-subfeatures

Files conflicting: x-pack/solutions/security/plugins/security_solution/common/index.ts

dplumlee committed:
updates security replacedBy's to include exceptions subfeature

dplumlee committed:
guards endpoint list call in create rule

rylnd committed:
Merge branch 'main' into rules-rbac-new

dhurley14 committed:
set authz strings for exceptions api's to use lists authz strings so endpoint exceptions tests stop failing. Added a TODO for discussion on how to move forward with the eventual plan to make lists it's own feature

dhurley14 committed:
Merge remote-tracking branch 'rylnd/rules-rbac-new' into rules-exceptions-subfeatures

dhurley14 committed:
first attempt at 'patching' update function when only read authz fields are modified in ruleUpdate object

rylnd committed:
Merge branch 'main' into rules-rbac-new

 Conflicts:
	x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_tables.tsx
	x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_management/api/rules/bulk_actions/route.ts
	x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_management/api/rules/find_rules/route.ts
	x-pack/solutions/security/test/serverless/api_integration/test_suites/platform_security/authorization.ts

dplumlee committed:
Merge remote-tracking branch 'rylnd/rules-rbac-new' into rules-exceptions-subfeatures

dhurley14 committed:
utilizes two way diff to determine which fields are modified in update route, if exceptions list is the only field 'updated' then we can use the special patch function from alerting

dhurley14 committed:
Merge remote-tracking branch 'origin/rules-exceptions-subfeatures' into rules-exceptions-subfeatures

dhurley14 committed:
limit use case for exceptions comparison to the update route so we don't interfere with prebuilt rules customization logic

dhurley14 committed:
type check failed in test suite

rylnd commented:
This PR was based on (the now squashed/merged) #239634 . Closing in favor of the squashed/rebased #245722.
